docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -242,9 +242,224 @@ This guide will help you create a new Expectation Suite by profiling your data w
+          rules:
+            - rule_1
+              rule_specific_domain_builders:
+                - rule_1_domain_builder_1:
+                - rule_1_domain_builder_2:
+                ...
+              rule_specific_parameter_builders:
+                - rule_1_parameter_builder_1:
+                # e.g. reference columns from rule_1_domain_builder_1 to calculate metric on each of them (e.g. ``column.min`` metric)
+                - rule_2_parameter_builder_2:
+                ...
+              expectation_configuration_builders:
+                - expectation_1_builder:
+                # e.g. expect_column_values_to_be_between where domain is set via columns from one of the domains defined above and min/max set via parameters defined above. Can also contain if/then logic.
+
+
+        Here is a slightly more concrete example:
+
+        .. code-block:: yaml
+
+          global_profiler_parameters:
+            - global_param_1
+          rules:
+            my_rule_for_ids: # semantic type
+              domains:
+                - my_id_domain_1: # QUESTION: Should we allow for multiple domains in a rule? QUESTION: Can / should domains be defined outside of the scope of a specific rule?
+                  # columns of type ``integer`` with name ``id``
+                - my_id_domain_2:
+                  # columns of type ``integer`` with name suffix of ``_id``
+              parameter_builders:
+                - my_id_parameter_1_min:
+                  # min value of column, using domains above
+                - my_id_parameter_2_max:
+                  # max value of column, using domains above
+                - my_id_parameter_3_min_5_batches_ago:
+                  # min value of column from 5 batches ago, using domains above and using a ``batch_request`` modifier to index to ``-5`` batches.
+              expectation_configuration_builders:
+                - expectation: expect_column_values_to_be_between # Expectation name as string
+                  column: $domain.my_id_domain_1.column # All columns in referenced domain (QUESTION: Why ``domain_kwargs`` in examples e.g. ``$domain.domain_kwargs.column``?)
+                  min_value: $my_id_parameter_1_min.parameter.min # Reference to parameter defined above
+                  max_value: $my_id_parameter_2_max.parameter.min
+                - expectation: expect_column_values_to_be_between
+                  column: $domain.my_id_domain_2.column
+                  min_value: $my_id_parameter_3_min_5_batches_ago.parameter.min
+                  max_value: $global_param_1
+                - expectation: expect_column_values_to_not_be_null
+                  column: $domains.column # columns for all domains
+
+            my_rule_for_datetimes: # semantic type
+              domains:
+                - my_datetime_domain_1: # QUESTION: Should we allow for multiple domains in a rule? QUESTION: Can / should domains be defined outside of the scope of a specific rule?
```

Comment:
Potential answer to question: currently it seems that a rule on several domains corresponds to multiple rules. However, it is necessary to allow multiple domains unless we specify a multi-batch domain for multi-batch to work.